crates/types/src/lib.rs

```diff
@@ -203,7 +203,7 @@ entity_impl!(GlobalIndex);
 
 /// Index type of a linear memory (imported or defined) inside the WebAssembly module.
 #[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Debug, Serialize, Deserialize)]
-pub struct MemoryIndex(u32);
+pub struct MemoryIndex(pub u32);
```

Comment:
Instead of making this `pub` can access to the underlying index go through `.as_u32()`?

Reply:
Fixed.